src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -101,51 +97,42 @@ export class InstructorStudentListPageComponent implements OnInit {
             return acc;
           }, {});
 
-          const teams: StudentIndexedData = students.students.reduce((acc: StudentIndexedData, x: Student) => {
-            const term: string = x.teamName;
-            (acc[term] = acc[term] || []).push(x);
-            return acc;
-          }, {});
-
-          courseTab.stats = {
-            numOfStudents: students.students.length,
-            numOfSections: Object.keys(sections).length,
-            numOfTeams: Object.keys(teams).length,
-          };
-
           Object.keys(sections).forEach((key: string) => {
             const studentsInSection: Student[] = sections[key];
 
-            const data: StudentListStudentData[] = [];
+            const studentList: StudentListRowModel[] = [];
             studentsInSection.forEach((student: Student) => {
-              const studentData: StudentListStudentData = {
-                name : student.name,
-                status : student.joinState,
-                email : student.email,
-                team : student.teamName,
+              const studentData: StudentListRowModel = {
+                name: student.name,
+                status: student.joinState,
+                email: student.email,
+                team: student.teamName,
+                sectionName: key,
+                isAllowedToModifyStudent: false,
+                isAllowedToViewStudentInSection: false,
               };
-              data.push(studentData);
+              studentList.push(studentData);
             });
 
-            this.loadPrivilege(courseTab, key, data);
+            this.loadPrivilege(courseTab, key, studentList);
           });
         }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorMessage(resp.error.message); });
   }
 
   /**
    * Loads privilege of an instructor for a specified course and section.
    */
-  loadPrivilege(courseTab: CourseTab, sectionName: string, students: StudentListStudentData[]): void {
+  loadPrivilege(courseTab: CourseTab, sectionName: string, students: StudentListRowModel[]): void {
     this.instructorService.loadInstructorPrivilege({ sectionName, courseId: courseTab.course.courseId })
         .subscribe((instructorPrivilege: InstructorPrivilege) => {
-          const sectionData: StudentListSectionData = {
-            sectionName,
-            students,
-            isAllowedToViewStudentInSection : instructorPrivilege.canViewStudentInSections,
-            isAllowedToModifyStudent : instructorPrivilege.canModifyStudent,
-          };
-
-          courseTab.studentListSectionDataList.push(sectionData);
+          students.forEach((student: StudentListRowModel) => {
+            student.isAllowedToViewStudentInSection = instructorPrivilege.canViewStudentInSections;
+            student.isAllowedToModifyStudent = instructorPrivilege.canModifyStudent;
+          });
+
+          courseTab.studentList.push(...students);
+          courseTab.stats =
```

Comment:
Many redundant calculations are going here (though it will provide correct result eventually). I think the correct way to do it is to construct `StudentListRowModel` when all students are loaded. After that, we can begin to load privilege for each section.